### ISSUE TITLE
Being able to disable each option in indent rule

### ISSUE THREAD

**Issue by srknzl** [eslint/eslint#14167]
**What rule do you want to change?**
indent
**Does this change cause the rule to produce more or fewer warnings?**
It will cause fewer warnings
**How will the change be implemented? (New option, new default behavior, etc.)?**
Adding a new behaviour (I assume it does not exist since giving it causes validation error). We will be able to give string "off" to options it is not available. SwitchCase option, FunctionDecleration option's body parameter and FunctionExpression option's body parameter does not have it. Other options have off as an option.
 
**Please provide some example code that this change will affect:**

For each option I will show two examples where one of them is ok and other one is not ok
* SwitchCase

This is ok:
```js
/*eslint indent: ["error", 2, { "SwitchCase": 1 }]*/
switch(a){
  case "a":
    break;
  case "b":
    break;
}
```

This is not ok:
```js
/*eslint indent: ["error", 2, { "SwitchCase": 1 }]*/
switch(a){
case "a":
    break;
case "b":
    break;
}
```

* FunctionDeclaration

This is ok:
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1} }]*/
function foo() {
  bar();
}
```

This is not ok:
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1} }]*/
function foo() {
    bar();
}
```

* FunctionExpression

This is ok:
```js
/*eslint indent: ["error", 2, { "FunctionExpression": {"body": 1} }]*/
var foo = function() {
  bar();
}
```

This is not ok:
```js
/*eslint indent: ["error", 2, { "FunctionExpression": {"body": 1} }]*/
var foo = function() {
    bar();
}
```

**What does the rule currently do for this code?**

The rule checks if the indentation level is correct. 

**What will the rule do after it's changed?**

It will ignore these cases and won't lint them.

Examples:

* SwitchCase

This is ok:
```js
/*eslint indent: ["error", 2, { "SwitchCase": "off" }]*/
switch(a){
case "a":
    break;
case "b":
    break;
}
```
This is also ok:
```js
/*eslint indent: ["error", 2, { "SwitchCase": "off" }]*/
switch(a){
  case "a":
    break;
  case "b":
    break;
}
```
* FunctionDeclaration

This is ok:
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": "off"} }]*/
function foo() {
  bar();
}
```

This is also ok:
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": "off"} }]*/
function foo() {
    bar();
}
```

* FunctionExpression

This is ok:
```js
/*eslint indent: ["error", 2, { "FunctionExpression": {"body": "off"} }]*/
var foo = function() {
  bar();
}
```

This is also ok:
```js
/*eslint indent: ["error", 2, { "FunctionExpression": {"body": "off"} }]*/
var foo = function() {
    bar();
}
```
**Are you willing to submit a pull request to implement this change?**
Sure

**Comment by mdjermanovic** [comment]
Hi @srknzl, thanks for the issue!

I think this can be done by configuring the existing [ignoredNodes](https://eslint.org/docs/rules/indent#ignorednodes) option like this:

```
"ignoredNodes": ["SwitchCase", "FunctionDeclaration > BlockStatement.body", "FunctionExpression > BlockStatement.body"]
```

With this configuration, your examples are all valid. Please see [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCAyLCB7IFxuICAgIFwiaWdub3JlZE5vZGVzXCI6IFtcbiAgICAgICAgXCJTd2l0Y2hDYXNlXCIsIFxuICAgICAgICBcIkZ1bmN0aW9uRGVjbGFyYXRpb24gPiBCbG9ja1N0YXRlbWVudC5ib2R5XCIsXG4gICAgICAgIFwiRnVuY3Rpb25FeHByZXNzaW9uID4gQmxvY2tTdGF0ZW1lbnQuYm9keVwiXG4gICAgXVxufV0qL1xuXG5zd2l0Y2goYSl7XG5jYXNlIFwiYVwiOlxuICAgIGJyZWFrO1xuY2FzZSBcImJcIjpcbiAgICBicmVhaztcbn1cblxuXG5zd2l0Y2goYSl7XG4gIGNhc2UgXCJhXCI6XG4gICAgYnJlYWs7XG4gIGNhc2UgXCJiXCI6XG4gICAgYnJlYWs7XG59XG5cbmZ1bmN0aW9uIGZvbygpIHtcbiAgYmFyKCk7XG59XG5cbmZ1bmN0aW9uIGZvbygpIHtcbiAgICBiYXIoKTtcbn1cblxudmFyIGZvbyA9IGZ1bmN0aW9uKCkge1xuICBiYXIoKTtcbn1cblxudmFyIGZvbyA9IGZ1bmN0aW9uKCkge1xuICAgIGJhcigpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=).

We had to make a decision to freeze the stylistic rules (you can find more details in the [blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)), so we won't be adding new options to the `indent` rule, but I believe that `ignoredNodes` already provides this functionality.

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).

**Comment by srknzl** [comment]
@mdjermanovic Many thanks for this.

 Just to give you feedback, I think the documentation about ignoreNodes can be improved. I did not understand that ignoreNodes can be used for this by reading short description of it: 
```
"ignoredNodes" accepts an array of selectors. If an AST node is matched by any of the selectors, the indentation of tokens which are direct children of that node will be ignored. This can be used as an escape hatch to relax the rule if you disagree with the indentation that it enforces for a particular syntactic pattern.
```
Thus, not moving to detailed description at all.

**Comment by mdjermanovic** [comment]
> I think the documentation about ignoreNodes can be improved.

Agreed, PRs to improve the documentation are always welcome!
